apps/trackers/jira/query.py

```diff
@@ -156,6 +156,11 @@ def generate_labels(self):
         if self.tracker.is_contract_priority:
             self._query["fields"]["labels"].append("contract-priority")
 
+        if self.tracker.is_compliance_priority:
+            # excluding the low priority ones
+            if Impact.LOW != self.tracker.aggregated_impact:
```

Comment:
I am thinking about one more thing. Should the impact check be really here or rather in the `is_compliance_priority` method? This way we split the business logic instead of keeping it all at one place. Or do you see a reason why the split is better/necessary?

Reply:
Looking at SFM2, is_compliance_priority is used for SLA calculations and for compliance-priority label. The latter has this LOW impact check. SLA calculations for compliance_priority are done differently for each impact. So moving it to the `is_compliance_priority` method would break any sane attempt at using it for SLA calculations, IMO.

As for _why_, I have no idea why in SFM2 the SLAs are that way nor why the LOW check is there when adding the compliance-priority label.

Reply:
I agree it is confusing there as the SLA is different for each impact by its nature. But SFM2 is actually a good example. In the SLA definition table at
https://gitlab.cee.redhat.com/prodsec-dev/sfm2/-/blob/master/sfm2/sla.py?ref_type=heads#L55
you can see that unlike the other impact values the LOW has the same SLA for both compliance and non-compliance priority. That basically means that the impact is part of the compliance priority check and LOW means that there is no special compliance priority handling. I would therefore rather put the impact check directly to the compliance priority check.

Reply:
I think that splitting the logic in SFM2 is actually a bad idea. I am for merging the impact and the product definitions checks under `is_compliance_priority`

Reply:
Fixed. Could you please do a final review on the last commit?

Thank you for the explanation. I have seen this code but didn't _actually see_ what you pointed out. Of course your suggestion makes perfect sense.